src/jit/lowerxarch.cpp

```diff
@@ -794,6 +794,42 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
         // the addr of SIMD vector with the given index.
         simdNode->gtOp1->gtFlags |= GTF_IND_REQ_ADDR_IN_REG;
     }
+    else if (simdNode->IsSIMDEqualityOrInequality())
+    {
+        LIR::Use simdUse;
+
+        if (BlockRange().TryGetUse(simdNode, &simdUse) && simdUse.User()->OperIs(GT_EQ, GT_NE) &&
+            simdUse.User()->gtGetOp2()->IsCnsIntOrI())
+        {
```

Comment:
The above condition mirrors the original code but it appears to be incomplete because it doesn't check that there are no nodes between JTRUE and SIMD (except the relop and its second op) that may change flags.